server/src/main/java/org/opensearch/action/admin/cluster/repositories/put/PutRepositoryRequest.java

```diff
@@ -67,12 +70,22 @@ public class PutRepositoryRequest extends AcknowledgedRequest<PutRepositoryReque
 
     private Settings settings = EMPTY_SETTINGS;
 
+    private Boolean encrypted;
+    private CryptoSettings cryptoSettings;
+
     public PutRepositoryRequest(StreamInput in) throws IOException {
         super(in);
         name = in.readString();
         type = in.readString();
         settings = readSettingsFromStream(in);
         verify = in.readBoolean();
+
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            encrypted = in.readOptionalBoolean();
+            if (Boolean.TRUE.equals(encrypted)) {
+                cryptoSettings = new CryptoSettings(in);
+            }
```

Comment:
Can we do `cryptoSettings = in.readOptionalWriteable(CryptoSettings::new)` and `out.writeOptionalWriteable(cryptoSettings)` instead of having an explicit extra boolean? For the user-facing API, it seems like it would be better to make the behavior that supplying crypto_settings makes the repo encrypted, and omitting them makes it unencrypted. Is there a reason to expose a separate boolean?

Reply:
Pushed the change.